lib/widgets/profile.dart

```diff
@@ -0,0 +1,334 @@
+        String? urlPattern = fieldData["url_pattern"];
+        if (urlPattern == null) {
+          final RealmDefaultExternalAccount? realmDefaultExternalAccount = store.realmDefaultExternalAccounts[fieldData["subtype"]];
+          if (realmDefaultExternalAccount != null) {
+            urlPattern = realmDefaultExternalAccount.urlPattern;
+          }
+        }
+        if (urlPattern != null) {
+          final url = urlPattern.replaceFirst("%(username)s", entry.text);
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: ProfileDataTableLink(url: url, label: entry.text));
+        } else {
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: Text(entry.text));
+        }
+      case CustomProfileFieldType.user:
+        // TODO(server): This is completely undocumented.  The key to
+        //   reverse-engineering it was:
+        //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+        final List<dynamic> userIds = jsonDecode(entry.text);
+        if (userIds.isEmpty) {
+          return const SizedBox.shrink();
+        }
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: userIds.whereType<num>()
+            .map((userId) => store.users[userId])
+            .whereType<User>()
+            .map((user) => ProfileDataTableUser(user: user)).toList());
+      case CustomProfileFieldType.date:
+      // TODO(server): The value's format is undocumented, but empirically
+      //   it's a date in ISO format, like 2000-01-01.
+      // That's readable as is, but:
+      // TODO format this date using user's locale.
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+      case CustomProfileFieldType.unknown:
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+    }
+  }
+}
+
+class CustomProfileFieldDisplayEntry {
+  final String label;
+  final String text;
+  final CustomProfileFieldType type;
+  final String fieldData;
+
+  CustomProfileFieldDisplayEntry({required this.fieldData, required this.label, required this.text, required this.type});
+}
+
+class ProfileDataTableLink extends StatelessWidget {
+  const ProfileDataTableLink({super.key, required this.label, required this.url});
+
+  final String label;
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    final Uri? parsedUrl = Uri.tryParse(url);
+    if (parsedUrl == null) {
+      return Text(label);
+    }
+    return MouseRegion(
+      cursor: SystemMouseCursors.click,
+      child: GestureDetector(
+        onTap: () async {
+          await ZulipBinding.instance.launchUrl(parsedUrl);
```

Comment:
At the other call site of this `launchUrl` method, we have some additional logic to adjust the UX:
```dart
    launched = await ZulipBinding.instance.launchUrl(url,
      mode: switch (Theme.of(context).platform) {
        // TODO(#279): On Android we settle for LaunchMode.externalApplication
        //   because url_launcher's in-app is a weirdly bare UX.
        //   Switch once that's fixed upstream (by us or otherwise).
        TargetPlatform.android => UrlLaunchMode.externalApplication,
        _ => UrlLaunchMode.platformDefault,
```

I think we'll definitely want that logic here too, for the same reasons.

In the future we'll also probably have a setting to choose whether to open in-app or externally (at least on iOS, and then on Android too once that mentioned issue is fixed), and we'll also want that to apply to both these places.

That other call site, in a `_launchUrl` function, also has a nice error dialog if for some reason we can't open the link; that'd also be nice to get here too. And if the URL string is relative, it resolves it against the realm's base URL. That's probably not common here, but again is probably for the best to have here too ­— if someone does put a relative URL string in one of these custom profile fields, then on web that will probably work great, and so someone may do that and expect it to work generally.

So in short I guess we should reuse the whole `_launchUrl` function. Perhaps just rename it to `launchContentUrl`? Given the realm-base-URL behavior, I think "launches a URL found in content from the Zulip server" is probably a good way to think about it, and so `content.dart` is not a bad home.